web-share/share-empty.https.html

```diff
@@ -0,0 +1,34 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <title>WebShare Test: Share no known fields</title>
+    <script src="/resources/testharness.js"></script>
+    <script src="/resources/testharnessreport.js"></script>
+    <script src="resources/manual-helper.js"></script>
+  </head>
+  <body>
+    <script>
+        promise_test(t => {
+          return promise_rejects(t, new TypeError(), navigator.share());
```

Comment:
Yessss. This was changed due to a bug filed on the Web Share issue tracker:
 [#47](https://github.com/WICG/web-share/issues/47), that Web IDL strictly mandates that if you have a dictionary parameter where all fields of the dictionary are optional, the parameter must be optional. Even if semantically it makes no sense because there are no situations in which passing nothing will do anything useful.

I argued a bit but then gave up. There's a [discussion from a year ago](https://github.com/heycam/webidl/issues/130) as precedent. It was decided to change existing specs by adding `optional` even where it makes no sense, rather than updating Web IDL.

There's a [note in the spec](https://wicg.github.io/web-share/#navigator-interface) highlighting that even though it says "optional" it actually isn't.